bson 0.15 build error!!!

+1 for this on macOS Catalina with rustc 1.41.0:

```
error[E0599]: no associated item named `INFINITY` found for type `f64` in the current scope
   --> /Users/alex/.cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/bson.rs:639:60
    |
639 |                 Ok("Infinity") => return Bson::Double(f64::INFINITY),
    |                                                            ^^^^^^^^ associated item not found in `f64`
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use core::num::dec2flt::rawfp::RawFloat;`
help: you are looking for the module in `std`, not the primitive type
    |
639 |                 Ok("Infinity") => return Bson::Double(std::f64::INFINITY),
    |                                                       ^^^^^^^^^^^^^^^^^^
```

+1 for windows 10 x64 with rustc 1.44.0

<details>
  <summary>Expand error message</summary>
  
```
error[E0599]: no associated item named `INFINITY` found for type `f64` in the current scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/bson.rs:639:60
    |
639 |                 Ok("Infinity") => return Bson::Double(f64::INFINITY),
    |                                                            ^^^^^^^^ associated item not found in `f64`
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use core::num::dec2flt::rawfp::RawFloat;`
help: you are looking for the module in `std`, not the primitive type
    |
639 |                 Ok("Infinity") => return Bson::Double(std::f64::INFINITY),
    |                                                       ^^^^^^^^^^^^^^^^^^

error[E0599]: no associated item named `NEG_INFINITY` found for type `f64` in the current scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/bson.rs:640:61
    |
640 |                 Ok("-Infinity") => return Bson::Double(f64::NEG_INFINITY),
    |                                                             ^^^^^^^^^^^^ associated item not found in `f64`
    |
help: you are looking for the module in `std`, not the primitive type
    |
640 |                 Ok("-Infinity") => return Bson::Double(std::f64::NEG_INFINITY),
    |                                                        ^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no associated item named `NAN` found for type `f64` in the current scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/bson.rs:641:55
    |
641 |                 Ok("NaN") => return Bson::Double(f64::NAN),
    |                                                       ^^^ associated item not found in `f64`
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use core::num::dec2flt::rawfp::RawFloat;`
help: you are looking for the module in `std`, not the primitive type
    |
641 |                 Ok("NaN") => return Bson::Double(std::f64::NAN),
    |                                                  ^^^^^^^^^^^^^

error[E0599]: no associated item named `MAX` found for type `u32` in the current scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/bson.rs:695:63
    |
695 | ...                   if t >= 0 && i >= 0 && t <= (u32::MAX as i64) && i <= (u32::MAX as i64)
    |                                                         ^^^ associated item not found in `u32`
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use rand::distributions::weighted::alias_method::Weight;`
help: you are looking for the module in `std`, not the primitive type
    |
695 |                             if t >= 0 && i >= 0 && t <= (std::u32::MAX as i64) && i <= (u32::MAX as i64)
    |                                                          ^^^^^^^^^^^^^

error[E0599]: no associated item named `MAX` found for type `u32` in the current scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/bson.rs:695:89
    |
695 | ...                   if t >= 0 && i >= 0 && t <= (u32::MAX as i64) && i <= (u32::MAX as i64)
    |                                                                                   ^^^ associated item not found in `u32`
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use rand::distributions::weighted::alias_method::Weight;`
help: you are looking for the module in `std`, not the primitive type
    |
695 |                             if t >= 0 && i >= 0 && t <= (u32::MAX as i64) && i <= (std::u32::MAX as i64)
    |                                                                                    ^^^^^^^^^^^^^

error[E0599]: no associated item named `MIN` found for type `i32` in the current scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/extjson/de.rs:188:34
    |
188 |                     if i >= i32::MIN as i64 && i <= i32::MAX as i64 {
    |                                  ^^^ associated item not found in `i32`
    |
help: you are looking for the module in `std`, not the primitive type
    |
188 |                     if i >= std::i32::MIN as i64 && i <= i32::MAX as i64 {
    |                             ^^^^^^^^^^^^^

error[E0599]: no associated item named `MAX` found for type `i32` in the current scope
   --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/extjson/de.rs:188:58
    |
188 |                     if i >= i32::MIN as i64 && i <= i32::MAX as i64 {
    |                                                          ^^^ associated item not found in `i32`
    |
    = help: items from traits can only be used if the trait is in scope
    = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
            `use rand::distributions::weighted::alias_method::Weight;`
help: you are looking for the module in `std`, not the primitive type
    |
188 |                     if i >= i32::MIN as i64 && i <= std::i32::MAX as i64 {
    |                                                     ^^^^^^^^^^^^^

error[E0599]: no associated item named `INFINITY` found for type `f64` in the current scope
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/extjson/models.rs:59:35
   |
59 |             "Infinity" => Ok(f64::INFINITY),
   |                                   ^^^^^^^^ associated item not found in `f64`
   |
   = help: items from traits can only be used if the trait is in scope
   = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
           `use core::num::dec2flt::rawfp::RawFloat;`
help: you are looking for the module in `std`, not the primitive type
   |
59 |             "Infinity" => Ok(std::f64::INFINITY),
   |                              ^^^^^^^^^^^^^^^^^^

error[E0599]: no associated item named `NEG_INFINITY` found for type `f64` in the current scope
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/extjson/models.rs:60:36
   |
60 |             "-Infinity" => Ok(f64::NEG_INFINITY),
   |                                    ^^^^^^^^^^^^ associated item not found in `f64`
   |
help: you are looking for the module in `std`, not the primitive type
   |
60 |             "-Infinity" => Ok(std::f64::NEG_INFINITY),
   |                               ^^^^^^^^^^^^^^^^^^^^^^

error[E0599]: no associated item named `NAN` found for type `f64` in the current scope
  --> /usr/local/cargo/registry/src/github.com-1ecc6299db9ec823/bson-0.15.0/src/extjson/models.rs:61:30
   |
61 |             "NaN" => Ok(f64::NAN),
   |                              ^^^ associated item not found in `f64`
   |
   = help: items from traits can only be used if the trait is in scope
   = note: the following trait is implemented but not in scope; perhaps add a `use` for it:
           `use core::num::dec2flt::rawfp::RawFloat;`
help: you are looking for the module in `std`, not the primitive type
   |
61 |             "NaN" => Ok(std::f64::NAN),
   |                         ^^^^^^^^^^^^^

error: aborting due to 10 previous errors

For more information about this error, try `rustc --explain E0599`.
error: could not compile `bson`.
warning: build failed, waiting for other jobs to finish...
error: build failed
```
</details>

--- a/src/bson.rs
+++ b/src/bson.rs
@@ -636,9 +636,9 @@ impl Bson {
             }
 
             ["$numberDouble"] => match doc.get_str("$numberDouble") {
-                Ok("Infinity") => return Bson::Double(f64::INFINITY),
-                Ok("-Infinity") => return Bson::Double(f64::NEG_INFINITY),
-                Ok("NaN") => return Bson::Double(f64::NAN),
+                Ok("Infinity") => return Bson::Double(std::f64::INFINITY),
+                Ok("-Infinity") => return Bson::Double(std::f64::NEG_INFINITY),
+                Ok("NaN") => return Bson::Double(std::f64::NAN),
                 Ok(other) => {
                     if let Ok(d) = other.parse() {
                         return Bson::Double(d);
@@ -692,7 +692,7 @@ impl Bson {
 
                     if let Ok(t) = timestamp.get_i64("t") {
                         if let Ok(i) = timestamp.get_i64("i") {
-                            if t >= 0 && i >= 0 && t <= (u32::MAX as i64) && i <= (u32::MAX as i64)
+                            if t >= 0 && i >= 0 && t <= (std::u32::MAX as i64) && i <= (std::u32::MAX as i64)
                             {
                                 return Bson::Timestamp(Timestamp {
                                     time: t as u32,

--- a/src/bson.rs
+++ b/src/bson.rs
@@ -692,7 +692,10 @@ impl Bson {
 
                     if let Ok(t) = timestamp.get_i64("t") {
                         if let Ok(i) = timestamp.get_i64("i") {
-                            if t >= 0 && i >= 0 && t <= (std::u32::MAX as i64) && i <= (std::u32::MAX as i64)
+                            if t >= 0
+                                && i >= 0
+                                && t <= (std::u32::MAX as i64)
+                                && i <= (std::u32::MAX as i64)
                             {
                                 return Bson::Timestamp(Timestamp {
                                     time: t as u32,

--- a/src/extjson/de.rs
+++ b/src/extjson/de.rs
@@ -185,7 +185,7 @@ impl TryFrom<serde_json::Value> for Bson {
             serde_json::Value::Number(x) => x
                 .as_i64()
                 .map(|i| {
-                    if i >= i32::MIN as i64 && i <= i32::MAX as i64 {
+                    if i >= std::i32::MIN as i64 && i <= std::i32::MAX as i64 {
                         Bson::Int32(i as i32)
                     } else {
                         Bson::Int64(i)

--- a/src/extjson/models.rs
+++ b/src/extjson/models.rs
@@ -56,9 +56,9 @@ pub(crate) struct Double {
 impl Double {
     pub(crate) fn parse(self) -> extjson::de::Result<f64> {
         match self.value.as_str() {
-            "Infinity" => Ok(f64::INFINITY),
-            "-Infinity" => Ok(f64::NEG_INFINITY),
-            "NaN" => Ok(f64::NAN),
+            "Infinity" => Ok(std::f64::INFINITY),
+            "-Infinity" => Ok(std::f64::NEG_INFINITY),
+            "NaN" => Ok(std::f64::NAN),
             other => {
                 let d: f64 = other.parse().map_err(|_| {
                     extjson::de::Error::invalid_value(

